File: code/sample.fs
// F# Sample
module Sample

let greet name = printfn "Hello, %s!" name

let factorial n =
    [1..n] |> List.fold (*) 1

[<EntryPoint>]
let main argv =
    greet "World"
    printfn "5! = %d" (factorial 5)
    0

File: code/sample.ml
(* OCaml Sample *)
let greet name =
  Printf.printf "Hello, %s!\n" name

let factorial n =
  let rec aux acc = function
    | 0 -> acc
    | n -> aux (acc * n) (n - 1)
  in aux 1 n

let () =
  greet "World";
  Printf.printf "5! = %d\n" (factorial 5)
